dot/network/status_test.go

```diff
@@ -109,5 +113,141 @@ func TestStatus(t *testing.T) {
+	msgRec = make(chan Message)
+	msgSend = make(chan Message)
+
+	// same for all network tests use the createTestService helper method
+	cfg.BlockState = blockState // required
+	cfg.NetworkState = &MockNetworkState{}
+	cfg.ProtocolID = TestProtocolID // default "/gossamer/dot/0"
+
+	node, err := NewService(cfg, msgSend, msgRec)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = node.Start()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	return node, msgSend, msgRec
+}
+
+// have a peer send a message status with a block ahead
+// test exchanged messages after peer connected are correct
+func TestStatusAndBlockRequestMessage(t *testing.T) {
+	dataDirA := path.Join(os.TempDir(), "gossamer-test", "nodeA")
+	defer os.RemoveAll(dataDirA)
+
+	configA := &Config{
+		DataDir:     dataDirA,
+		Port:        7001,
+		RandSeed:    1,
+		NoBootstrap: true,
+		NoMdns:      true,
+	}
+
+	blockState := newMockBlockState(big.NewInt(3))
+	nodeA, msgSendA, msgRecA := createTestServiceWithBlockState(t, configA, blockState)
+	defer nodeA.Stop()
+
+	nodeA.noGossip = true
+
+	genesisHash, err := common.HexToHash("0xdcd1346701ca8396496e52aa2785b1748deb6db09551b72159dcb3e08991025b")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	bestBlockHash, err := common.HexToHash("0x829de6be9a35b55c794c609c060698b549b3064c183504c18ab7517e41255569")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	testStatusMessage := &StatusMessage{
+		ProtocolVersion:     uint32(2),
+		MinSupportedVersion: uint32(2),
+		Roles:               byte(4),
+		BestBlockNumber:     uint64(2434417),
+		BestBlockHash:       bestBlockHash,
+		GenesisHash:         genesisHash,
+		ChainStatus:         []byte{0},
+	}
+
+	// simulate host status message sent from core service on startup
+	msgRecA <- testStatusMessage
+
+	dataDirB := path.Join(os.TempDir(), "gossamer-test", "nodeB")
+	defer os.RemoveAll(dataDirB)
+
+	configB := &Config{
+		DataDir:     dataDirB,
+		Port:        7002,
+		RandSeed:    2,
+		NoBootstrap: true,
+		NoMdns:      true,
+	}
+
+	blockState = newMockBlockState(big.NewInt(1))
+	nodeB, _, msgRecB := createTestServiceWithBlockState(t, configB, blockState)
+	defer nodeB.Stop()
+
+	nodeB.noGossip = true
+
+	// simulate host status message sent from core service on startup
+	msgRecB <- testStatusMessage
+
+	addrInfosB, err := nodeB.host.addrInfos()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = nodeA.host.connect(*addrInfosB[0])
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	time.Sleep(TestStatusTimeout)
+
+	if !nodeA.status.confirmed(nodeB.host.h.ID()) {
+		t.Error("node A did not confirm status of node B")
+	}
+
+	if !nodeB.status.confirmed(nodeA.host.h.ID()) {
+		t.Error("node B did not confirm status of node A")
+	}
+
+	// get latest block header from block state
+	latestHeader := blockState.LatestHeader()
+	currentHash := blockState.LatestHeader().Hash()
+
+	// expected block request message
+	var ThisTestMessage = &BlockRequestMessage{
```

Comment:
cane we remove `var` and name `expected` or `expectedMessage`?

Reply:
yes